webpack.dev-server.config.js

```diff
@@ -0,0 +1,69 @@
+const client = require('./src/client/webpack.config');
+const server = require('./src/server/webpack.config');
+const { HotModuleReplacementPlugin } = require('webpack');
+
+const {
+  SERVER_PORT = 3000,
+} = process.env;
+
+/**
+ * Watch settings are the same between client and server, so we're keeping them
+ * here for consistency
+ */
+const watchConfig = {
+  watchOptions: {
+    ignored: /node_modules/,
+  },
+};
+
+/**
+* Webpack Dev Server Config
+*
+* This webpack configuration somewhat mimics `webpack.config.js`, but also
+* provides live reloading for the client side code and hot module replacement
+* for the server-side code
+*
+* @param {object} env Webpack `env` object
+*/
+module.exports = ({ mode = 'development' } = {}) => ([
+  {
+    ...client.config({ mode }, process.env),
```

Comment:
The dev-server also needs some extra config under `resolve`:

```js
    alias: {
      'react-dom': '@hot-loader/react-dom',
    },
```

as in https://github.com/seas-computing/react-boilerplate/blob/master/src/server/middleware/dev.middleware.ts#L49-L51.

As well as the `react-hot-loader` plugin under `modules`:

```js
        use: [
          {
            loader: 'babel-loader',
            options: {
              plugins: ['react-hot-loader/babel'],
            },
          },
          'ts-loader',
```

https://github.com/seas-computing/react-boilerplate/blob/master/src/server/middleware/dev.middleware.ts#L35-42

Reply:
Whoops! I must've missed those when I copied the client config over...